eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -32,59 +42,93 @@ function Get-SwaggerInfo()
         Write-Error $_
     }
     Write-Host "Cannot find swagger info"
-    Pop-Location
     exit 1
 }
 
-function Update-AutorestConfigFile() {
+function CreateOrUpdateAutorestConfigFile() {
     param (
         [string]$autorestFilePath,
         [string]$inputfile = "",
-        [string]$readme = ""
+        [string]$readme = "",
+        [string]$autorestConfigYaml = ""
     )
-    if (Test-Path -Path $autorestFilePath) {
-        if ($readme -ne "") {
-            Write-Host "Updating autorest.md file to config required readme file."
+
+    if ((Test-Path -Path $autorestFilePath) -and (![String]::IsNullOrWhiteSpace((Get-Content -Path $autorestFilePath)))) {
+        if (($readme -ne "") -or ($inputfile -ne "")) {
             $requirRex = "require*:*";
-            $inputfileRex = "input-file *:*"
-            $requirefile = $readme + [Environment]::NewLine + "- " + $readme.Replace("readme.md", "readme.csharp.md")
-            if ((Get-Content $autorestFilePath | Select-String -Pattern $requirRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.md" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -notmatch $inputfileRex |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                (Get-Content $autorestFilePath) -replace $requirRex, ("require:" + [Environment]::NewLine + "- " + "$requirefile") | Set-Content $autorestFilePath
-            } elseif ((Get-Content $autorestFilePath | Select-String -Pattern $inputfileRex).Matches.Success) {
-                (Get-Content $autorestFilePath) -notmatch "- .*.json" |Out-File $autorestFilePath
-                $requirefile = $requirefile + [Environment]::NewLine + "csharp: true";
-                (Get-Content $autorestFilePath) -replace $inputfileRex, ("require:" + [Environment]::NewLine + "- " + "$requirefile") | Set-Content $autorestFilePath
+            $inputfileRex = "input-file*:*"
+            $fileContent = Get-Content -Path $autorestFilePath
+            # clear
+            $fileContent = $fileContent -notmatch $requirRex
```

Comment:
I'm not sure I understand what these are doing? Are you trying to strip out parts of the file? Should we instead switch to a model that just pulls out what we want with a regex instead?

Reply:
I need to update the autorest.md from required readme to input-file json file, and vice versa. In order to support different scenarios, the input-file / require may define in one line, or use `-` to several lines (see following example), I need to clear all the previous lines and add the new lines. 
```
input-file: https://*****/**.json
```
or 
```
input-file:
- https://****/**.json
```

Reply:
A similar regex can be used if you just want to replace similar to:
```
$fileContent = $fileContent -replace "(?s)(input-file:[\s-]*)(?<url>.+)","`$1$inputFile"
```

You can do something similar for the other settings.

Reply:
We need to support the scenario which will contain more than one line (see senario 2), and it also switch between input-file and require. The path of the file can be url or file path. So it is not be able to use regex and replace to support all those scenarios.

Reply:
My replace statement also supports the multiple line case that is what the `(?s)` option is for. If you test that regex locally it should handle both cases. Even if it isn't a url the regex should work for those cases.

Reply:
If we read the file content without `-Raw`  `$filecontent = (Get-Content -path autorest.md)`, it will return an string array, one line on string, then the regex cannot handle multiple lines scenario.
If we read the file content with `-Raw` `$filecontent = (Get-Content -path autorest.md -Raw)`, it will return an string. The regex will return all the content after 'input-file' line (the input-file and other content behind)  which is not what we want, and cannot replace it.

Reply:
You can fix the regex such that it doesn't return the rest of the file by adding `(?s)(input-file:[\s-]*)(?<url>\S+)` which says match all the non-whitespace characters.

Reply:
@chunyu3 I'd still like for you to try out my regex for replacement as I think it can make this code a lot simpler.

Reply:
Thanks @weshaggard I will try your Regex. The problem I met before is that in order to match multiple lines, I need to Get-content with `-Raw`, in that case the all content will be read as a string. And the problem is that it will return all the content starting from the first match line which is not what we want. I will retry and see if any Regex can resolve this issue and only return the matched lines only. Thanks

Reply:
Hello @weshaggard I defined an Regex which can match out all scenarios and use `-replace` to update the autorest.md. Please take a look. Thanks.